src/reducers/wasteSettingsReducer.ts

```diff
@@ -42,16 +35,7 @@ export enum WasteSettingsActions {
 type WasteSettingsAction =
   | { type: WasteSettingsActions.setActiveType; payload: { key: string; value: boolean } }
   | {
-      type: WasteSettingsActions.setActiveTypesForOldStreet;
-      payload: {
-        [key: string]: {
-          active: boolean;
-          storeId?: number | string;
-        };
-      };
-    }
-  | {
-      type: WasteSettingsActions.resetActiveTypesWithOldStreetCleanup;
+      type: WasteSettingsActions.resetActiveTypes;
```

Comment:
If I have already saved a street and I want to update it, as a user it would be nice to see a loading icon on the screen after pressing the “save” button. Because depending on the internet connection, requests can be slow and I don't see any changes on the screen, so I thought that the “save” button wasn't working and pressed it several times in a row. Then I found myself in HomeScreen. Because many times the `goBack()` function worked.

Reply:
Not only a visual feedback for the user. You cannot press the save multiple times, which was your issue.

Reply:
Yes. That's how we can solve both problems. 👍